fastestimator/backend/save_model.py

```diff
@@ -64,7 +65,12 @@ def save_model(model: Union[tf.keras.Model, torch.nn.Module],
             assert model.current_optimizer, "optimizer does not exist"
             optimizer_path = os.path.join(save_dir, "{}_opt.pkl".format(model_name))
             with open(optimizer_path, 'wb') as f:
-                pickle.dump({'weights': model.current_optimizer.get_weights(), 'lr': get_lr(model)}, f)
+                saved_data = {'weights': model.current_optimizer.get_weights(), 'lr': get_lr(model)}
+                if isinstance(model.current_optimizer, tfa.optimizers.DecoupledWeightDecayExtension) or hasattr(
```

Comment:
if we're going to use a hasattr check should we just replace all of this with a hasattr(weight_decay) ?